src/plugins/data/common/search/search_source/legacy/call_client.ts

```diff
@@ -36,7 +36,8 @@ import { ISearchOptions } from '../../index';
 export function callClient(
   searchRequests: SearchRequest[],
```

Comment:
Does the search request have the index pattern ? Or data source info?

Reply:
The search requests interface is not being updated. The data source ID is only present on the client side. It was added at [line 359](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/0605a80acd8b9d1dfb461bba72e26a9cc3ed94bf/src/plugins/data/common/search/search_source/search_source.ts#L359) and removed at [line 86](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/0605a80acd8b9d1dfb461bba72e26a9cc3ed94bf/src/plugins/data/common/search/search_source/legacy/fetch_soon.ts#L86). The msearch API's search requests interface remains unchanged. The only modification to the internal msearch API was the addition of a new `data_source_id` query parameter. This parameter will trigger a `msearch` call on the specified data source.